staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -64,6 +65,7 @@ func StorageWithCacher() generic.StorageDecorator {
 			IndexerFuncs:   triggerFuncs,
 			Indexers:       indexers,
 			Codec:          storageConfig.Codec,
+			Clock:          clock.RealClock{},
```

Comment:
Let's not set it.
Use the smart defaulting in cacher itself (if Clock is nil, the default to RealClock).

Same in the test.